src/e2e/java/teammates/e2e/cases/e2e/InstructorStudentListPageE2ETest.java

```diff
@@ -0,0 +1,131 @@
+package teammates.e2e.cases.e2e;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.attributes.CourseAttributes;
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.util.AppUrl;
+import teammates.common.util.Const;
+import teammates.e2e.pageobjects.InstructorCourseEnrollPage;
+import teammates.e2e.pageobjects.InstructorCourseStudentDetailsEditPage;
+import teammates.e2e.pageobjects.InstructorCourseStudentDetailsViewPage;
+import teammates.e2e.pageobjects.InstructorStudentListPage;
+import teammates.e2e.pageobjects.InstructorStudentRecordsPage;
+
+/**
+ * SUT: {@link Const.WebPageURIs#INSTRUCTOR_STUDENT_LIST_PAGE}.
+ */
+public class InstructorStudentListPageE2ETest extends BaseE2ETestCase {
+
+    @Override
+    protected void prepareTestData() {
+        testData = loadDataBundle("/InstructorStudentListPageE2ETest.json");
+        removeAndRestoreDataBundle(testData);
+    }
+
+    @Test
+    public void testAll() {
+
+        ______TS("verify loaded data");
+
+        InstructorAttributes instructor = testData.instructors.get("instructorOfCourse1");
+        String instructorId = instructor.googleId;
+
+        AppUrl listPageUrl = createUrl(Const.WebPageURIs.INSTRUCTOR_STUDENT_LIST_PAGE).withUserId(instructorId);
+        InstructorStudentListPage listPage = loginAdminToPage(listPageUrl, InstructorStudentListPage.class);
+
+        CourseAttributes course1 = testData.courses.get("course1");
+        CourseAttributes course2 = testData.courses.get("course2");
+        CourseAttributes course3 = testData.courses.get("course3");
+        String course1Header = createHeaderText(course1);
+        String course2Header = createHeaderText(course2);
+        String course3Header = createHeaderText(course3);
+
+        StudentAttributes[] studentsInCourse1 = {};
+
+        // Note: by right, there should not be any student shown here as the instructor does not have sufficient privilege
+        // However, due to issue #8000, the students will be listed anyway
```

Comment:
Should we use a instructor with the appropriate privilege instead?

Reply:
No; the entire point is to ensure that there should be nothing displayed here.

Reply:
Oh I see what you mean now.